src/server/admin/server/admin_test.go

```diff
@@ -192,33 +193,84 @@ func RepoInfoToName(repoInfo interface{}) interface{} {
 	return repoInfo.(*pfs.RepoInfo).Repo.Name
 }
 
-// testExtractRestore effectively implements both TestExtractRestoreObjects
-// TestExtractRestoreNoObjects, as their logic is mostly the same
-func testExtractRestore(t *testing.T, testObjects bool) {
+// testExtractRestore effectively implements TestExtractRestoreObjects,
+// TestExtractRestoreNoObjects, TestExtractRestoreAuth, as their logic is mostly the same
+func testExtractRestore(t *testing.T, testObjects, testAuth bool) {
 	if os.Getenv("RUN_BAD_TESTS") == "" {
 		t.Skip("Skipping because RUN_BAD_TESTS was empty")
 	}
 	if testing.Short() {
 		t.Skip("Skipping integration tests in short mode")
 	}
 
+	numPipelines := 3
 	c := tu.GetPachClient(t)
+	pipelineClients := make([]*client.APIClient, numPipelines)
```

Comment:
Initialize all the clients up front, because after much suffering I realized `GetAuthenticatedPachClient` resets the set of admins every time :|